### PR TITLE
Allow signers and verifiers to work on digests

### PR DESCRIPTION
**Pull request by loosebazooka**
- Add concrete verifier types, package up verifier logic

Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

**Comment by loosebazooka** [comment]
just some stuff I ran into when writing the final keyless verification flows

**Comment by patflynn** [comment]
could we just remove artifact bytes verification if digest based is the right way to go?

**Comment by loosebazooka** [comment]
Signers are used in a few other places. Like oidc subject signing

File: src/main/java/dev/sigstore/encryption/signers/RsaVerifier.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    this.publicKey = publicKey;
+  }
+
+  @Override
+  public PublicKey getPublicKey() {
+    return publicKey;
+  }
+
+  @Override
+  public boolean verify(byte[] artifact, byte[] signature)
```
Comment (patflynn) [inline src/main/java/dev/sigstore/encryption/signers/RsaVerifier.java:35]: these two methods look identical apart from the algo. maybe delegate to internal shared method?

Comment (loosebazooka) [reply src/main/java/dev/sigstore/encryption/signers/RsaVerifier.java:35]: Man I had them all as parameterized versions of the same class. But then this weirdo rsa hash prefix got me good

**Review by patflynn — APPROVED** [review]

**Pull request merged**